crates/eframe/src/native/file_storage.rs

```diff
@@ -56,21 +56,24 @@ impl FileStorage {
     /// Find a good place to put the files that the OS likes.
     pub fn from_app_id(app_id: &str) -> Option<Self> {
         crate::profile_function!(app_id);
-        if let Some(data_dir) = storage_dir(app_id) {
-            if let Err(err) = std::fs::create_dir_all(&data_dir) {
-                log::warn!(
-                    "Saving disabled: Failed to create app path at {:?}: {}",
-                    data_dir,
-                    err
-                );
+        storage_dir(app_id).map_or_else(
```

Comment:
This `map_or_else` doesn't make sense to me here, I think a better way would be to use guarded let-else to return early